app/uk/gov/hmrc/emcstfe/repositories/GetMovementRepository.scala

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.emcstfe.repositories
+
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model._
+import play.api.libs.json.Format
+import uk.gov.hmrc.emcstfe.config.AppConfig
+import uk.gov.hmrc.emcstfe.models.mongo.GetMovementMongoResponse
+import uk.gov.hmrc.emcstfe.models.response.ErrorResponse
+import uk.gov.hmrc.emcstfe.utils.{Logging, TimeMachine}
+import uk.gov.hmrc.mongo.MongoComponent
+import uk.gov.hmrc.mongo.play.json.PlayMongoRepository
+import uk.gov.hmrc.mongo.play.json.formats.MongoJavatimeFormats
+
+import java.time.Instant
+import java.util.concurrent.TimeUnit
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class GetMovementRepository @Inject()(
+                                                    mongoComponent: MongoComponent,
+                                                    appConfig: AppConfig,
+                                                    time: TimeMachine
+                                                  )(implicit ec: ExecutionContext)
+  extends PlayMongoRepository[GetMovementMongoResponse](
+    collectionName = "get-movement-response",
+    mongoComponent = mongoComponent,
+    domainFormat   = GetMovementMongoResponse.format,
+    indexes        = Seq(
+      IndexModel(
+        Indexes.ascending("lastUpdated"),
+        IndexOptions()
+          .name("lastUpdatedIdx")
+          .expireAfter(appConfig.getMovementTTL().toSeconds, TimeUnit.SECONDS)
+      ),
+      IndexModel(
+        Indexes.compoundIndex(
+          Indexes.ascending("internalId"),
+          Indexes.ascending("ern"),
+          Indexes.ascending("arc")
+        ),
+        IndexOptions().name("uniqueIdx")
+      )
+    ),
+    replaceIndexes = true // TODO check if this deleted data on startup or not
```

Comment:
I don't think this would delete any data on startup - it would just replace the indexes